src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java

```diff
@@ -0,0 +1,48 @@
+package com.github.princesslana.eriscasper.data.immutable.encoding;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.Optional;
+import org.immutables.encode.Encoding;
+import org.immutables.encode.Encoding.StandardNaming;
+
+@Encoding
+public class OptionalImmutableListEncoding<T> {
+  @Encoding.Impl private Optional<ImmutableList<T>> value = Optional.empty();
+
+  @Encoding.Builder
+  static class Builder<T> {
```

Comment:
We should heavily consider having tests for these to make sure the encoding is working. Compile -> Test so we don't need to worry about the compilation procedure. I think it would be a good idea to test the encoding just like we do the Nullable class and Optional differences.